ql/src/java/org/apache/hadoop/hive/ql/stats/ColStatsProcessor.java

```diff
@@ -232,6 +236,19 @@ public int persistColumnStats(Hive db, Table tbl) throws HiveException, MetaExce
   public void setDpPartSpecs(Collection<Partition> dpPartSpecs) {
   }
 
+  private void setOrRemoveColumnStatsAccurateProperty(Hive db, TableName tableName, List<String> colNames, boolean success) throws HiveException {
+    EnvironmentContext environmentContext = new EnvironmentContext();
+    environmentContext.putToProperties(StatsSetupConst.DO_NOT_UPDATE_STATS, StatsSetupConst.TRUE);
+    // Fetch latest table object
+    Table newTable = db.getTable(tableName.getDbTable());
```

Comment:
why do you need this? don't you already have a tbl instance?

Reply:
what if we switch back to `metastore` source, would the stats be still accurate?

Reply:
> why do you need this? don't you already have a tbl instance?

The tbl instance is modified by BasicStatsTask to include Basic stats flag, which isn't present in the tbl provided by ColStatsProcessor, hence we need to fetch the latest tbl object.

> what if we switch back to metastore source, would the stats be still accurate?

Nice catch, I didnt consider this and saw that this was required for both the cases (native/non-native table). I have updated the patch to set COLUMN_STATS_ACCURATE flag in general. Done.

Reply:
> > why do you need this? don't you already have a tbl instance?
> 
> The tbl instance is modified by BasicStatsTask to include Basic stats flag, which isn't present in the tbl provided by ColStatsProcessor, hence we need to fetch the latest tbl object.
> 
> > what if we switch back to metastore source, would the stats be still accurate?
> 
> Nice catch, I didnt consider this and saw that this was required for both the cases (native/non-native table). I have updated the patch to set COLUMN_STATS_ACCURATE flag in general. Done.

won't it require an extra db call? could we clone the object before it gets modified?

Reply:
@deniskuzZ We require the modified object itself in this case, hence this requires a table fetch from the metastore.

Reply:
@deniskuzZ I have modified the code to avoid doing this DB call so that we use the result table object of BasicStatsTask.